base/linalg/diagonal.jl

```diff
@@ -225,6 +225,9 @@ A_mul_B!(A::AbstractMatrix,B::Diagonal)  = scale!(A,B.diag)
 A_mul_Bt!(A::AbstractMatrix,B::Diagonal) = scale!(A,B.diag)
 A_mul_Bc!(A::AbstractMatrix,B::Diagonal) = scale!(A,conj(B.diag))
 
+Ac_mul_B(A::Diagonal,B::AbstractVector) = conj.(A.diag) .* B
```

Comment:
Our `(c)transpose` is recursive so it should be `ctranspose.(A.diag) .* B`. It would be great if you could make the same fix for `Ac/t_mul_B!`.